numba/core/entrypoints.py

```diff
@@ -20,5 +21,11 @@ def init_all():
 
     for entry_point in iter_entry_points('numba_extensions', 'init'):
         logger.debug('Loading extension: %s', entry_point)
-        func = entry_point.load()
-        func()
+        try:
+            func = entry_point.load()
+            func()
+        except Exception as e:
+            msg = "Numba extension module '{}' failed to load due to '{}({})'."
+            warnings.warn(msg.format(entry_point.module_name, type(e).__name__,
+                                     str(e)), stacklevel=2)
+            logger.debug('Extension loading failed for: %s', entry_point)
```

Comment:
```suggestion
            logger.debug('Extension loading failed for: %s', entry_point, exc_info=True)
```

Reply:
Thanks for the suggestion, however this PR is merged, please feel free to open a new PR with the suggested change. Thanks!